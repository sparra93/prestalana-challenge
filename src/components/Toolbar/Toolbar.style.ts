import styled from 'styled-components';

export const ToolbarContainer = styled.header`
  display: flex;
  padding: 10px 20px;
  justify-content: space-between;
  align-items: center;
  background: rgb(59, 92, 162);
  background: radial-gradient(
    circle,
    rgba(59, 92, 162, 1) 11%,
    rgba(59, 92, 162, 1) 100%,
    rgba(2, 0, 36, 0.8799894957983193) 100%
  );
`;

export const ToolbarTitle = styled.h1`
  color: #fff;
  font-size: 1.5em;
  margin: 0;
`;

export const CartIconContainer = styled.div`
  display: flex;
  align-items: center;
  position: relative;
  cursor: pointer;
`;

export const CartIcon = styled.img`
  width: 24px;
  height: 24px;
  margin-right: 10px;
`;

export const Badge = styled.div`
  background-color: red;
  color: white;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 0.8em;
  position: absolute;
  top: -5px;
  right: 0;
`;
