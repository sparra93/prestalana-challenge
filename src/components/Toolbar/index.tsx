import Link from 'next/link';
import { useSelector } from 'react-redux';

import {
  Badge,
  CartIcon,
  CartIconContainer,
  ToolbarContainer,
  ToolbarTitle,
} from '@/components/Toolbar/Toolbar.style';
import { selectCartCountTotal } from '@/redux/features/productSlice';

const Toolbar: React.FC = () => {
  const productCount = useSelector(selectCartCountTotal);

  return (
    <ToolbarContainer>
      <ToolbarTitle>My Personal Dashboard</ToolbarTitle>
      <Link href="/cart" passHref>
        <CartIconContainer>
          <CartIcon src="/assets/icons/shopping-bag.png" alt="Cart Icon" />
          <Badge>{productCount}</Badge>
        </CartIconContainer>
      </Link>
    </ToolbarContainer>
  );
};

export default Toolbar;
