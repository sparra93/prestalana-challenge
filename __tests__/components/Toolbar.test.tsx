import { screen } from '@testing-library/react';
import React from 'react';

import Toolbar from '@/components/Toolbar';

describe('Toolbar Component', () => {
  it('should display toolbar information correctly', () => {
    renderWithProviders(<Toolbar />);
    expect(screen.getByAltText('Cart Icon')).toHaveAttribute(
      'src',
      '/assets/icons/shopping-bag.png',
    );
  });

  it('should have a icon bag that redirect to /cart page', () => {
    renderWithProviders(<Toolbar />);

    const link = screen.getByTestId('go-to-cart');
    expect(link).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProviders(<Toolbar />);

    expect(asFragment()).toMatchSnapshot();
  });
});
