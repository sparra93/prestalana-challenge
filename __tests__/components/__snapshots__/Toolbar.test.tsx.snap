// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="sc-ece30d93-0 hPavrx"
  >
    <h1
      class="sc-ece30d93-1 dQbGoR"
    >
      My Personal Dashboard
    </h1>
    <a
      data-testid="go-to-cart"
      href="/Cart"
    >
      <div
        class="sc-ece30d93-2 ejPOtw"
      >
        <img
          alt="Cart Icon"
          class="sc-ece30d93-3 cwysBr"
          src="/assets/icons/shopping-bag.png"
        />
        <div
          class="sc-ece30d93-4 ekOwio"
        >
          0
        </div>
      </div>
    </a>
  </header>
</DocumentFragment>
`;
